File: core/focused_ux.py
"""Progressive, presentation-only task navigation for public and admin pages."""
import re
VERSION='20260923-focused'
ASSETS='/design-system/studio/'

def decorate_focused_ux(text, *, home=False):
    if '</head>' not in text:return text
    text=re.sub(r'<link\b[^>]*id=[\"\x27]focused-ux-style[\"\x27][^>]*>','',text)
    text=re.sub(r'<script\b[^>]*id=[\"\x27]focused-ux-script[\"\x27][^>]*>\s*</script>','',text)
    tags=f'<link id="focused-ux-style" rel="stylesheet" href="{ASSETS}focused-ux.css?v={VERSION}"><script id="focused-ux-script" defer src="{ASSETS}focused-ux.js?v={VERSION}"></script>'
    text=re.sub(r'\s*</head>',lambda _:tags+'\n</head>',text,count=1)
    if home:
        # The main headline and existing consultation/menu actions remain.
        text=re.sub(r'<aside\b[^>]*class=[\"\x27][^\"\x27]*\bhero-advantage\b[^\"\x27]*[\"\x27][^>]*>.*?</aside>','',text,flags=re.S)
    return text

def admin_hub():
    groups=[
      ('create','ブログ制作','/admin/apps/blog','記事の下書きをつくる','記事 文章 画像'),
      ('create','ブログ管理','/admin/blog','記事を編集・公開する','記事 一覧 編集 公開'),
      ('create','リール制作','/admin/apps/reel/','動画と投稿文をつくる','動画 Instagram'),
      ('create','SNS投稿','/admin/sns-post','下書きと投稿を確認する','Threads X 発信'),
      ('check','SNS分析','/admin/gubble-sns','反応から次の一手へ','数値 分析 Instagram'),
      ('check','公開ページ','/','公開中の表示を見る','サイト トップ'),
      ('organize','実行指令室','/admin/command-center','予定と指示をまとめる','予定 タスク カレンダー'),
      ('organize','AI相談','/admin/chat','考えと次の行動を整理する','チャット メモ'),
      ('organize','OPS','/ops','資料とプロンプトを探す','資料 手順 プロンプト'),
      ('check','相場羅針盤','/admin/command-center/trade','調査・計画・記録を見る','相場 市場 株 取引'),
    ]
    from html import escape
    cards=''.join(f'<a class="ux-task-card" href="{url}" data-ux-category="{group}" data-ux-keywords="{escape(keywords)}"><span><strong>{title}</strong><small>{desc}</small></span><span aria-hidden="true">↗</span></a>' for group,title,url,desc,keywords in groups)
    return f'''<main class="ux-admin-home" id="main-content"><header class="ux-admin-intro"><p>AI相談 / WORKSPACE</p><h1>今日は、何を進めますか。</h1></header>
<section class="ux-task-browser" aria-label="管理作業を選ぶ"><div class="ux-task-controls" hidden><label class="ux-search-label" for="ux-task-search">作業を探す<input id="ux-task-search" type="search" placeholder="記事、動画、予定…" autocomplete="off"></label><div class="ux-filters" role="group" aria-label="管理作業の目的"><button type="button" data-ux-filter="all" aria-pressed="true">すべて</button><button type="button" data-ux-filter="create" aria-pressed="false">つくる</button><button type="button" data-ux-filter="check" aria-pressed="false">確かめる</button><button type="button" data-ux-filter="organize" aria-pressed="false">整える</button></div></div><div class="ux-task-grid">{cards}</div><p class="ux-empty" hidden>見つかりませんでした。別の言葉で探すか、「すべて」に戻してください。</p><p class="ux-result-status" role="status" aria-live="polite"></p></section>
<footer class="ux-admin-footer"><a href="/">公開ページを見る ↗</a><a href="/admin/logout">ログアウト</a></footer></main>'''

def replace_admin_hub(text):
    if 'ux-admin-home' in text:return decorate_focused_ux(text)
    text,count=re.subn(r'<main\b[^>]*>.*?</main>',lambda _:admin_hub(),text,count=1,flags=re.S)
    assert count==1,'Expected admin main'
    return decorate_focused_ux(text)


def decorate_admin_entry(text):
    # One protected asset also serves blog subroutes: retain all existing forms.
    if 'data-ux-admin-hub' not in text:
        hub=admin_hub().replace('<main ', '<main hidden data-ux-admin-hub ', 1)
        text,count=re.subn(r'(<div class="container">)',lambda m:hub+m[0],text,count=1)
        assert count==1,'Expected protected admin container'
    return decorate_focused_ux(text)

File: site/static/design-system/studio/focused-ux.js
/* Local navigation only: no generation, network requests, or input storage. */
(() => {
 'use strict';
 if (!document.body.classList.contains('studio-editorial') || document.body.classList.contains('studio-login') || document.body.dataset.focusedUx) return;
 document.body.dataset.focusedUx='true';
 const normalize=value=>String(value).normalize('NFKC').toLocaleLowerCase().trim();
 const matches=(text,query)=>normalize(query).split(/\s+/).every(word=>normalize(text).includes(word));
 const make=(tag,cls,text)=>{const el=document.createElement(tag);if(cls)el.className=cls;if(text)el.textContent=text;return el;};

 // Keep every course in the HTML; show a smaller choice only after enhancement.
 const courseList=document.querySelector('#course-voices');
 if(courseList){
  const cards=[...courseList.querySelectorAll('.compact-course-card')];
  if(cards.length===6){
   const groups=[['all','すべて'],['learn','学ぶ・試す'],['team','組織で使う'],['build','制作を任せる']];
   const bar=make('div','ux-filters');bar.setAttribute('role','group');bar.setAttribute('aria-label','コースを目的で絞る');
   const status=make('p','ux-result-status');status.setAttribute('role','status');
   const categories=['learn','learn','learn','team','learn','build'];
   groups.forEach(([id,label])=>{const button=make('button','',label);button.type='button';button.setAttribute('aria-pressed',String(id==='all'));
    button.addEventListener('click',()=>{bar.querySelectorAll('button').forEach(b=>b.setAttribute('aria-pressed',String(b===button)));cards.forEach((card,i)=>{card.hidden=id!=='all'&&categories[i]!==id;});status.textContent=`${label}：${cards.filter(c=>!c.hidden).length}件`;});bar.append(button);
   });
   courseList.before(bar);courseList.after(status);
   // Direct links must never land on a filtered-out course.
   window.addEventListener('hashchange',()=>{let id;try{id=decodeURIComponent(location.hash.slice(1));}catch{return;}const target=document.getElementById(id);const card=target?.closest('.compact-course-card');if(card?.hidden){bar.querySelector('button').click();target.scrollIntoView({block:'start'});}});
  }
 }
 const adminHub=document.querySelector('[data-ux-admin-hub]');
 if(adminHub && /^\/admin\/?$/.test(location.pathname)){adminHub.hidden=false;const original=document.querySelector('body > .container');if(original){original.hidden=true;original.classList.add('ux-original-admin');}}
 const taskBrowser=document.querySelector('.ux-task-browser');
 if(taskBrowser){
  const controls=taskBrowser.querySelector('.ux-task-controls'),search=taskBrowser.querySelector('input'),cards=[...taskBrowser.querySelectorAll('.ux-task-card')],buttons=[...taskBrowser.querySelectorAll('[data-ux-filter]')];
  let category='all';
  const render=()=>{cards.forEach(card=>{card.hidden=!(category==='all'||card.dataset.uxCategory===category)||!matches(card.textContent+' '+card.dataset.uxKeywords,search.value);});const count=cards.filter(c=>!c.hidden).length;taskBrowser.querySelector('.ux-empty').hidden=count!==0;taskBrowser.querySelector('.ux-result-status').textContent=`${count}件の作業`;buttons.forEach(b=>b.setAttribute('aria-pressed',String(b.dataset.uxFilter===category)));};
  buttons.forEach(button=>button.addEventListener('click',()=>{category=button.dataset.uxFilter;if(category==='all')search.value='';render();}));
  search.addEventListener('input',()=>{category='all';render();});controls.hidden=false;render();
 }

 const header=document.querySelector('.site-header-inner');
 if(!header || typeof HTMLDialogElement==='undefined')return;
 const admin=document.body.classList.contains('studio-admin')||document.body.classList.contains('admin-page');
 let commands;
 if(admin){
  const unique=new Map();
  document.querySelectorAll('.admin-shared-header a[href],.ux-task-card').forEach(a=>{const href=a.getAttribute('href');if(!href?.startsWith('/')||href.startsWith('//')||href.includes('logout'))return;if(!unique.has(href))unique.set(href,{href,label:(a.querySelector('strong,h3')?.textContent||a.textContent).trim().replace(/\s+/g,' ')});});
  commands=[...unique.values()];
 }else commands=[{label:'講習・相談コース',href:'/#packages'},{label:'AI実力診断',href:'/ai-agent-readiness/'},{label:'サイト診断',href:'/seo-llmo-diagnosis/'},{label:'AIニュース・Codex',href:'/ai-news/'},{label:'AIアプリサイト制作',href:'/ai-app-site/'},{label:'受講資料',href:'/lectures/index.html'},{label:'ブログ',href:'/blog/'},{label:'講師紹介',href:'/speaker.html'},{label:'実績サイト',href:'/#all-works'},{label:'困っている仕事を相談する',href:'/#contact'}];
 if(!commands.length)return;
 const launch=make('button','ux-launcher','探す');launch.type='button';launch.setAttribute('aria-haspopup','dialog');launch.setAttribute('aria-label',admin?'管理作業を探す':'目的のページを探す');
 const dialog=make('dialog','ux-command-dialog');dialog.setAttribute('aria-labelledby','ux-command-title');
 const top=make('div','ux-command-top');const title=make('h2','',admin?'どの作業へ進みますか？':'どこから始めますか？');title.id='ux-command-title';const close=make('button','ux-command-close','閉じる');close.type='button';top.append(title,close);
 const label=make('label','ux-search-label','キーワードで探す');const input=make('input');input.type='search';input.placeholder=admin?'記事、SNS、予定…':'講習、診断、資料…';input.autocomplete='off';label.append(input);
 const list=make('div','ux-command-results');const empty=make('p','ux-empty','見つかりませんでした。別の言葉で探してください。');empty.hidden=true;
 const count=make('p','ux-result-status');count.setAttribute('role','status');
 const links=commands.map(item=>{const a=make('a','',item.label);a.href=item.href;a.addEventListener('click',()=>dialog.close());list.append(a);return a;});
 dialog.append(top,label,list,empty,count);document.body.append(dialog);header.append(launch);
 let previous;
 const filter=()=>{links.forEach(a=>a.hidden=!matches(a.textContent,input.value));const visible=links.filter(a=>!a.hidden).length;empty.hidden=visible!==0;count.textContent=`${visible}件`;};
 const open=()=>{if(dialog.open||document.querySelector('dialog[open]'))return;previous=document.activeElement;input.value='';filter();dialog.showModal();input.focus();};
 launch.addEventListener('click',open);close.addEventListener('click',()=>dialog.close());input.addEventListener('input',filter);
 input.addEventListener('keydown',event=>{if(event.key==='ArrowDown'){event.preventDefault();links.find(a=>!a.hidden)?.focus();}});
 dialog.addEventListener('keydown',event=>{if(event.key==='Escape'){event.preventDefault();event.stopPropagation();dialog.close();}});
 dialog.addEventListener('close',()=>{if(previous?.isConnected)previous.focus();});
 document.addEventListener('keydown',event=>{if((event.ctrlKey||event.metaKey)&&!event.altKey&&event.key.toLowerCase()==='k'&&!document.querySelector('dialog[open]')){event.preventDefault();open();}});
})();
